src/odb/src/db/dbInsertBuffer.cpp

```diff
@@ -0,0 +1,1685 @@
+        logger_->error(
+            utl::ODB,
+            1206,
+            "Cannot create hierarchical connection: '{}' is not a descendant "
+            "of '{}'.",
+            load_pin->getInst()->getName(),
+            target_module->getName());
+        break;
+      }
+
+      // 1. Check if the current net is reusable (using cache)
+      if (tryReuseParentPath(
+              load_obj, current_module, top_mod_iterm, load_pins)) {
+        continue;
+      }
+
+      // 2. Check if there is ANY other ModNet in this module that we can reuse.
+      if (tryReuseModNetInModule(load_obj, current_module, top_mod_iterm)) {
+        continue;
+      }
+
+      // 3. Create ports & modnet
+      createNewHierConnection(
+          load_obj, current_module, top_mod_iterm, base_name);
+    }
+  }
+
+  // 4. Perform connections
+  performFinalConnections(load_pin, drvr_term, top_mod_iterm);
+
+  return top_mod_iterm != nullptr;
+}
+
+dbModNet* dbInsertBuffer::getFirstModNetInFaninOfLoads(
+    const std::set<dbObject*>& load_pins,
+    const std::set<dbModNet*>& modnets_in_target_module) const
+{
+  for (dbObject* load : load_pins) {
+    dbModNet* curr_mod_net = nullptr;
+
+    // 1. Get the modnet of the load
+    if (load->getObjectType() == dbITermObj) {
+      dbITerm* iterm = static_cast<dbITerm*>(load);
+      curr_mod_net = iterm->getModNet();
+    } else if (load->getObjectType() == dbBTermObj) {
+      curr_mod_net = static_cast<dbBTerm*>(load)->getModNet();
+    } else {
+      continue;
+    }
+
+    if (curr_mod_net == nullptr) {
+      continue;
+    }
+
+    // 2. Find the modnet in the target module by fanin traversal
+    while (curr_mod_net) {
+      // Found the modnet in the target module
+      if (modnets_in_target_module.find(curr_mod_net)
+          != modnets_in_target_module.end()) {
+        return curr_mod_net;
+      }
+
+      // Go up to the next modnet in the fanin
+      curr_mod_net = curr_mod_net->getNextModNetInFanin();
+    }
+  }
+
+  return nullptr;
+}
+
+dbModNet* dbInsertBuffer::getFirstDriverModNetInTargetModule(
+    const std::set<dbModNet*>& modnets_in_target_module) const
+{
+  // 1. Find the driver terminal of this flat net
+  dbObject* driver_term = net_->getFirstDriverTerm();
+  if (driver_term == nullptr) {
+    return nullptr;
+  }
+
+  // 2. Get the driver's modnet
+  dbModNet* curr_modnet = nullptr;
+  if (driver_term->getObjectType() == dbITermObj) {
+    curr_modnet = static_cast<dbITerm*>(driver_term)->getModNet();
+  } else if (driver_term->getObjectType() == dbBTermObj) {
+    curr_modnet = static_cast<dbBTerm*>(driver_term)->getModNet();
+  }
+
+  // 3. Traverse fanout from driver's modnet to find target module's modnet
+  while (curr_modnet) {
+    // Found the first modnet in the target module
+    if (modnets_in_target_module.contains(curr_modnet)) {
+      return curr_modnet;
+    }
+    curr_modnet = curr_modnet->getNextModNetInFanout();
```

Comment:
I unclear on how this works - this search from one path forward.  If the module is in a different path it won't be found.